server/repository/src/db_diesel/changelog/test.rs

```diff
@@ -0,0 +1,572 @@
+        record_id: "invoice2".to_string(),
+        row_action: ChangelogAction::Upsert,
+        name_id: Some("name3".to_string()),
+        store_id: Some("store3".to_string()),
+    };
+
+    let log4 = ChangelogRow {
+        cursor: 4,
+        table_name: ChangelogTableName::StocktakeLine,
+        record_id: "stocktake_line1".to_string(),
+        row_action: ChangelogAction::Upsert,
+        name_id: None,
+        store_id: None,
+    };
+
+    for log in vec![&log1, &log2, &log3, &log4] {
+        diesel::insert_into(changelog_dsl::changelog)
+            .values(log)
+            .execute(&connection.connection)
+            .unwrap();
+    }
+
+    let repo = ChangelogRepository::new(&connection);
+
+    // Filter by table name
+
+    assert_eq!(
+        repo.changelogs(
+            0,
+            20,
+            Some(ChangelogFilter::new().table_name(ChangelogTableName::Requisition.equal_to()))
+        )
+        .unwrap(),
+        vec![log2.clone()]
+    );
+
+    // Filter by name_id in
+
+    assert_eq!(
+        repo.changelogs(
+            0,
+            20,
+            Some(ChangelogFilter::new().name_id(EqualFilter::equal_any(vec![
+                "name1".to_string(),
+                "name3".to_string()
+            ])))
+        )
+        .unwrap(),
+        vec![log1.clone(), log3.clone()]
+    );
+
+    // Filter by store_id in or null
+
+    assert_eq!(
+        repo.changelogs(
+            0,
+            20,
+            Some(
+                ChangelogFilter::new().store_id(EqualFilter::equal_any_or_null(vec![
+                    "store1".to_string(),
+                    "store2".to_string()
+                ]))
+            )
+        )
+        .unwrap(),
+        vec![log1.clone(), log2.clone(), log4.clone()]
+    );
+}
+
+struct TestRecord<T> {
+    record: T,
+    record_id: String,
+    name_id: String,
+    store_id: String,
+}
+
+/// Helper method to test name and store id
+/// Does db operation passed in as a function and then queries changelog to confirm name_id and store_id are set correctly
+fn test_changelog_name_and_store_id<T, F>(
+    connection: &StorageConnection,
+    record: TestRecord<T>,
+    row_action: ChangelogAction,
+    db_op: F,
+) where
+    F: Fn(&StorageConnection, &T),
+{
+    let repo = ChangelogRepository::new(&connection);
+
+    db_op(connection, &record.record);
+
+    let change_logs = repo
+        .changelogs(
+            0,
+            20,
+            Some(ChangelogFilter::new().record_id(EqualFilter::equal_to(&record.record_id))),
+        )
+        .unwrap();
+
+    assert_eq!(
+        change_logs[0],
+        inline_edit(&change_logs[0], |mut r| {
+            r.name_id = Some(record.name_id);
+            r.store_id = Some(record.store_id);
+            r.record_id = record.record_id;
+            r.row_action = row_action.clone();
+            r
+        })
+    );
+}
+
+#[actix_rt::test]
+async fn test_changelog_name_and_store_id_in_trigger() {
```

Comment:
It might be helpful to add a comment here something like this ...

```suggestion
async fn test_changelog_name_and_store_id_in_trigger() {

// This test checks that the database triggers that should create change log records are working correctly
// For each record type we create an example record, then check that the associated changelog record has automatically been created.
```

Reply:
Great, thank you, sorry missed that one, I did have a concern that these tests are confusing and was going to add extra comment, but forgot